### ISSUE TITLE
Reproducable assertion failure in 'package:flutter/src/widgets/scrollable.dart' from example app.

### ISSUE THREAD

**Issue by boustrophedon** [flutter/flutter#11689]
## Steps to Reproduce

In the examples/flutter_gallery app, open the animation demo. Select one of the cards so it is in the "fully expanded" state (sorry, I just started with flutter and mobile dev so I don't know what it's called).

Two finger scroll to the left or right with one finger in the header section (Eyeglasses/Seating/Decoration/Protection) and one finger in the content section "Flutter enables interactive animation". You should get the assertion failure in the title. The same assertion fails in one other bug, #11433.

I've worked on Dart before (I interned on the Fiber TV team!) and I want to learn flutter so if someone points me in the right direction I could try and work on this bug. I suppose src/widgets/scrollable.dart:425 would be the place to start?

## Logs

Run your application with `flutter run` and attach all the log output.
https://gist.github.com/boustrophedon/0d06378b8e05d74e154e26fa9e37486a

Run `flutter analyze` and attach any output of that command also.
https://gist.github.com/boustrophedon/47acd18f02b4395f7cc7977fc0773cf0

## Flutter Doctor

Paste the output of running `flutter doctor` here.
```
[✓] Flutter (on Linux, locale en_US.UTF-8, channel alpha)
    • Flutter at /home/hcs/code/flutter
    • Framework revision b156a0f054 (7 days ago), 2017-08-11 22:01:40 -0700
    • Engine revision fef7d827d6
    • Tools Dart version 1.25.0-dev.9.0

[✓] Android toolchain - develop for Android devices (Android SDK 26.0.1)
    • Android SDK at /home/hcs/code/Android/Sdk
    • Platform android-26, build-tools 26.0.1
    • ANDROID_HOME = /home/hcs/code/Android/Sdk
    • Java binary at: /opt/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] Android Studio (version 2.3)
    • Android Studio at /opt/android-studio
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] Connected devices
    • Pixel • FA6A30300472 • android-arm • Android 7.1.2 (API 25)
```
> For more information about diagnosing and reporting Flutter bugs, please see [https://flutter.io/bug-reports/](https://flutter.io/bug-reports/).

**Comment by jason-simmons** [comment]
The animation demo contains a header region and a details region.  Each region has a scroll notification listener that tries to keep the other region's scroll position in sync when you scroll horizontally.

It looks like the sequence of events is:
  * Tap in the details region.  The details ScrollableState gets a HandleDragDown callback and establishes a HoldScrollActivity.  
  * Tap in the header region.
  * Scroll horizontally in the header region.
    * The header's listener (_AnimationDemoHomeState._handlePageNotification) calls jumpToWithoutSettling to keep the details ScrollPosition in sync.
    * This disposes the details HoldScrollActivity, and the details ScrollableState clears its _hold ScrollHoldController.
  * Scroll horizontally in the details region.  The details ScrollableState gets a DragStart gesture.
    * But the ScrollableState's _hold was already cleared, triggering the assert failure.

@Hixie any ideas?  What is the best way to keep the ScrollableState coherent given that changes to its state are coming from both drag events and ScrollPosition API calls?

Would it be safe to have ScrollableState._handleDragStart ignore the event if the _hold was already disposed by the time the drag start arrives?

**Comment by Hixie** [comment]
I'd have to look more closely to be sure but my impression is that the approach of using scroll notification listeners in this way is fundamentally not going to work. Better to have them share a scroll position or generally act more like the LinkedScrollView (in the tests).

**Comment by Hixie** [comment]
cc @HansMuller 

keywords: LinkedScrollController LinkedScrollPosition

**Comment by Piinks** [comment]
Appears to be related to # 14452 as well

**Comment by Piinks** [comment]
Actually, this appears to have been fixed by #13164. 
The assert that is being triggered here does not exist anymore, and was removed with that change.

Inline documentation from [scrollable.dart](https://github.com/flutter/flutter/blob/9bb7142ef7183f3c31f2f62f977a73f3b0149615/packages/flutter/lib/src/widgets/scrollable.dart#L486):
> It's possible for _hold to become null between _handleDragDown and _handleDragStart, for example if some user code calls jumpTo or otherwise triggers a new activity to begin.

I've checked the gallery app to see if there was any more bad behavior, and I have not found any. 
Since non-existent asserts can not be triggered, I'm going to close this issue. 😉
